tools/toolchain/optimized_clang.sh

```diff
@@ -0,0 +1,113 @@
+
+if [[ "${CLANG_BUILD}" = "INSTALL" ]]; then
+    # Build a PGO-optimized compiler using the boostrapped compiler.
+    rm -rf "${CLANG_BUILD_DIR}"
+    git clone https://github.com/llvm/llvm-project --branch llvmorg-"${LLVM_CLANG_TAG}" --depth=1 "${CLANG_BUILD_DIR}"
+    cd "${CLANG_BUILD_DIR}"
+    cmake -B build -S llvm "${CLANG_OPTS[@]}" -DLLVM_BUILD_INSTRUMENTED=IR
+    ninja -C build
+
+    cd "${SCYLLA_DIR}"
+    rm -rf build_profile build_profile.ninja
+    ./configure.py "${SCYLLA_OPTS[@]}"
+    LLVM_PROFILE_FILE="${CLANG_BUILD_DIR}"/build/profiles/default_%p-%m.profraw ninja -f build_profile.ninja build
+
+    cd "${CLANG_BUILD_DIR}"
+    llvm-profdata merge "${CLANG_BUILD_DIR}"/build/profiles/default_*.profraw -output=ir.prof
+    rm -rf build
+    cmake -B build -S llvm "${CLANG_OPTS[@]}" -DLLVM_BUILD_INSTRUMENTED=CSIR -DLLVM_PROFDATA_FILE="$(realpath ir.prof)"
+    ninja -C build
+
+    # 2nd compilation: gathering a clang profile for CSPGO
+    cd "${SCYLLA_DIR}"
+    rm -rf build_profile build_profile.ninja
+    ./configure.py "${SCYLLA_OPTS[@]}"
+    LLVM_PROFILE_FILE="${CLANG_BUILD_DIR}"/build/profiles/csir-%p-%m.profraw ninja -f build_profile.ninja build
+
+    cd "${CLANG_BUILD_DIR}"
+    llvm-profdata merge build/csprofiles/default_*.profraw -output=csir.prof
+    llvm-profdata merge ir.prof csir.prof -output=combined.prof
+    rm -rf build
+    # -DLLVM_LIBDIR_SUFFIX=64 for Fedora compatibility
+    cmake -B build -S llvm "${CLANG_OPTS[@]}" -DLLVM_PROFDATA_FILE="$(realpath combined.prof)" -DCMAKE_EXE_LINKER_FLAGS="-Wl,--emit-relocs" -DCMAKE_INSTALL_PREFIX=/usr/local -DLLVM_LIBDIR_SUFFIX=64
+    ninja -C build
+
+    #TODO: skipping BOLT for aarch64 for now, since it causes segfault
+    if [[ "${ARCH}" != "aarch64" ]]; then
+        # BOLT phase
+        mv build/bin/clang-"${CLANG_SUFFIX}" build/bin/clang-"${CLANG_SUFFIX}".prebolt
+        mkdir -p build/profiles
+        llvm-bolt build/bin/clang-"${CLANG_SUFFIX}".prebolt -o build/bin/clang-"${CLANG_SUFFIX}" --instrument --instrumentation-file="${CLANG_BUILD_DIR}"/build/profiles/prof --instrumentation-file-append-pid --conservative-instrumentation
+
+        # 3rd ScyllaDB compilation: gathering a clang profile for BOLT
+        cd "${SCYLLA_DIR}"
+        rm -rf build_profile build_profile.ninja
+        ./configure.py "${SCYLLA_OPTS[@]}"
+        ninja -f build_profile.ninja build
+
+        cd "${CLANG_BUILD_DIR}"
+        merge-fdata build/profiles/*.fdata > prof.fdata
+        llvm-bolt build/bin/clang-"${CLANG_SUFFIX}".prebolt -o build/bin/clang-"${CLANG_SUFFIX}" --data=prof.fdata --reorder-functions=hfsort --reorder-blocks=ext-tsp --split-functions --split-all-cold --split-eh --dyno-stats
+    fi
+
+    cd "${CLANG_ROOT_DIR}"
+    rm -rf "${CLANG_BUILD_DIR}"/{build/profiles,*.prof,prof.fdata}
+    tar -cpzf "${CLANG_ARCHIVE}" "${CLANG_CHECKOUT_NAME}"
```

Comment:
How much excess data does this contain? Just the clang sources?

I think it's acceptable, trying to remove them from the archive would be difficult.

Reply:
Profiles file size are:
```
$ du -sch "${CLANG_BUILD_DIR}"/{build/profiles,*.prof,prof.fdata}
467G    clang_build/llvm-project/build/profiles
80M     clang_build/llvm-project/combined.prof
54M     clang_build/llvm-project/csir.prof
46M     clang_build/llvm-project/ir.prof
263M    clang_build/llvm-project/prof.fdata
467G    total
```
Total build/ directory size is 470GB, so almost all file size are profiles (467GB) and rest of them are only 3GB.
So I think it's better to drop profiles when create tar archive.